### ISSUE TITLE
Chatcraft crashes when trying to access shared chats

### ISSUE THREAD

**Issue by Rachit1313** [tarasglek/chatcraft.org#480]
<img width="1427" alt="image" src="https://github.com/tarasglek/chatcraft.org/assets/58596563/9a37003a-aec8-4e5e-a735-8af40f400af0">

Console error:
```json
{status: 404, statusText: 'Not Found', internal: true, data: 'Error: No route matches URL "/api/share/Rachit1313/juWInjaLnUTnbj9ApOdTe"', error: Error: No route matches URL "/api/share/Rachit1313/juWInjaLnUTnbj9ApOdTe"
    at Do (https://chatcr…}
    ```

**Comment by Rachit1313** [comment]
previously encountered this issue in #446

**Comment by Rachit1313** [comment]
Triage Meeting outcomes:
1. The issue is with /api/share router.
2. When accessing /api/share from browser it redirects to /c/chatId which correctly renders the shared that but this logic doesn't work when we are trying to access /api/share from react-router-dom i.e the shared chats sidebar it is looking up for the api/share URL and not entering the loadShare() function which redirects it to correct URL.

**Comment by Rachit1313** [comment]
@rjwignar , Any progress on this?

**Comment by WangGithub0** [comment]
@Rachit1313  @rjwignar  Is there any progress on this issue？

**Comment by Rachit1313** [comment]
> @Rachit1313 @rjwignar Is there any progress on this issue？

tbh not as of now.. still trying to debug it.

**Comment by rjwignar** [comment]
@Rachit1313 @Amnish04 
After investigating this the past couple of days, I can confidently say I've found the issue:

We know the **shared chats actually exist**, because "/api/share/{user}/{id}" **links work if you copy/paste the link** in your address bar, and redirects to the `/c/{user}/{id}`-style url.

**But** for whatever reason (possibly react-router-dom incompatibility), the "/api/share/{user}/{id}" doesn't work when used as a redirect in the Sidebar.

### What's the solution?
Previously, Shared Chats were given the `/c/{user}/{id}`-style URL upon creation, which would be the Chat's redirect link in the UI.
This is why Shared Chats created a month ago could still be directly accessed from the UI, while recent Shared Chats could not.

It's not the best, but I proposed one fix in #506, which gives Shared chats the `/c/{user}/{id}`-style URL upon creation instead of the `/api/share/{user}/{id}`.

Ideally, I want to find out why the `/api/share/{user}/{id}`-style URLs work when you copy/paste them in your browser but don't work as a redirect-URL in the UI.
